### PR TITLE
WIP Hdf5 Code Generation: VLValToBuf, VLBufToVal

### PR DESCRIPTION
**Pull request by hodger**
This PR generates the last two functions in hdf5_back.cc, VLValToBuf and VLBufToVal. I am seeing all unit tests passing. WIP because there are a couple small things left to clean up in other functions.

This also handles generating the header file declarations for VLValToBuf, as well as a few bug fixes to previous generation work. The main fix checks if data passed to the WriteToBuf function exceeds the length of the column. If so, we remove elements from the end of the container to compensate.

### CODE REVIEW

**Comment by scopatz** [comment]
This looks great to me @hodger! Please ping when this is no longer a work in progress. I think it could be merged in as-is.

**Comment by scopatz** [comment]
Can you please merge from master.

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -23,6 +23,17 @@ CONFIGURE_FILE(
...
     )
 
+EXECUTE_PROCESS(COMMAND python ${CMAKE_CURRENT_SOURCE_DIR}/hdf5_back_gen.py "VAL_TO_BUF_H" OUTPUT_VARIABLE HDF5_BACK_CC_VAL_TO_BUF_H RESULT_VARIABLE res_var_h)
+IF(NOT "${res_var_h}" STREQUAL "0")
+  message(FATAL_ERROR "Process hdf5_back_gen.py 'VAL_TO_BUF_H' failed, result = '${res_var_h}'")
+ENDIF()
+
+CONFIGURE_FILE(
+    "${CMAKE_CURRENT_SOURCE_DIR}/hdf5_back.h.in"
+    "${CMAKE_CURRENT_SOURCE_DIR}/hdf5_back.h"
```
Comment (scopatz) [inline src/CMakeLists.txt]: After merging from master, can you please update this to match the *.gen* pattern the hdf5back.cc uses.

**Comment by scopatz** [comment]
@aglenn2 please review this too. @hodger, would you mind getting together and walking him through the code base. I think that would be nice.

**Comment by hodger** [comment]
@scopatz sure thing. we can try to find some time this week/weekend

**Comment by hodger** [comment]
@aglenn2 I added those comments I mentioned earlier, so the code should be a bit more readable.

**Comment by scopatz** [comment]
Is this good to go?

**Comment by hodger** [comment]
@scopatz I believe that it is. I don't think @aglenn2 has been added as a Cyclus member, so he may not have gotten the notification that I tagged him above

**Comment by scopatz** [comment]
Thanks @hodger! Great work!

**Pull request merged**